toplevel/coqinit.ml

```diff
@@ -75,8 +82,14 @@ let libs_init_load_path ~coqlib =
   let coq_path = Names.DirPath.make [Libnames.coq_root] in
 
   (* ML includes *)
-  let plugins_dirs = System.all_subdirs ~unix_path:(coqlib/"plugins") in
-  List.map fst plugins_dirs,
+  let plugins_dirs = System.all_subdirs ~unix_path:(coqlib/"plugins") |> List.map fst in
+
+  let contrib_ml, contrib_vo = build_userlib_path ~unix_path:user_contrib in
+
+  let misc_ml, misc_vo =
+    List.map (fun s -> build_userlib_path ~unix_path:s) (xdg_dirs @ coqpath) |> List.split in
+
+  plugins_dirs @ contrib_ml @ List.concat misc_ml,
```

Comment:
I don't like having such a big expression after the comma, could we take advantage of changing this code to let-lift a bit?

Reply:
Sure, but that makes the diff quite worse tho :/

Reply:
Adding in another commit kinda helps.